WalletWasabi.Tests/RegressionTests/WabiSabi/WabiSabiApiApplicationFactory.cs

```diff
@@ -7,92 +7,64 @@
 using Microsoft.AspNetCore.Mvc.Testing;
 using Microsoft.AspNetCore.TestHost;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
 using NBitcoin;
-using NBitcoin.RPC;
 using WalletWasabi.BitcoinCore.Rpc;
 using WalletWasabi.Crypto.Randomness;
 using WalletWasabi.Tests.Helpers;
-using WalletWasabi.Tests.UnitTests;
 using WalletWasabi.WabiSabi.Backend;
 using WalletWasabi.WabiSabi.Backend.Banning;
 using WalletWasabi.WabiSabi.Backend.PostRequests;
 using WalletWasabi.WabiSabi.Backend.Rounds;
 using WalletWasabi.WabiSabi.Client;
 using WalletWasabi.WabiSabi.Crypto;
+using WalletWasabi.WabiSabi.Models.MultipartyTransaction;
 
 namespace WalletWasabi.Tests.RegressionTests.WabiSabi
 {
-	public class WabiSabiApiApplicationFactory : WebApplicationFactory<WalletWasabi.Backend.Startup>
+	public class WabiSabiApiApplicationFactory<TStartup> : WebApplicationFactory<TStartup> where TStartup : class
 	{
-		public WabiSabiApiApplicationFactory()
-			: base()
+		protected override IHostBuilder CreateHostBuilder()
 		{
+			var builder = Host.CreateDefaultBuilder().ConfigureWebHostDefaults(x =>
+			{
+				x.UseStartup<TStartup>().UseTestServer();
+			});
+			return builder;
 		}
 
 		protected override void ConfigureWebHost(IWebHostBuilder builder)
 		{
 			// will be called after the `ConfigureServices` from the Startup
 			builder.ConfigureTestServices(services =>
 			{
-				services.AddSingleton<IRPCClient>(GetMockRpc());
-				services.AddTransient<IArenaRequestHandler, ArenaRequestHandler>();
-				services.AddTransient<Prison>();
-				services.AddTransient<WabiSabiConfig>();
-				services.AddSingleton<Arena>(serviceProvider =>
-				{
-					var rpc = serviceProvider.GetRequiredService<IRPCClient>();
-					var prison = serviceProvider.GetRequiredService<Prison>();
-					var config = serviceProvider.GetRequiredService<WabiSabiConfig>();
-					var arena = new Arena(TimeSpan.FromSeconds(4), Network.Main, config, rpc, prison);
-					arena.StartAsync(CancellationToken.None).GetAwaiter().GetResult();
-					arena.TriggerAndWaitRoundAsync(TimeSpan.FromSeconds(1)).GetAwaiter().GetResult();
-					return arena;
-				});
+				services.AddHostedService<BackgroundServiceStarter<Arena>>();
+				services.AddSingleton<Arena>();
+				services.AddSingleton<ArenaRequestHandler>();
+				services.AddScoped<Network>(_ => Network.Main);
+				services.AddScoped<IRPCClient>(_ => BitcoinFactory.GetMockMinimalRpc());
+				services.AddScoped<Prison>();
+				services.AddScoped<WabiSabiConfig>();
+				services.AddScoped(typeof(TimeSpan), _ => TimeSpan.FromSeconds(2));
```

Comment:
This `Startup` class is for setting up the testing http server only. It has nothing to do with the  `WalletWasabi.Backend.Setup` class that is the one that runs in real world. However it is really useful to understand what has to be setted and how, because we can see what dependencies are required, how to register hosted services, how to register the json converters and how to inject value types, and who knows what else we can have in the future.

Currently, instead of doing things in this way, that is how we should register all the things, we have a `WabiSabiCoordinator` class that does what we are doing here.

-----------------

Going back to your question: in real world we should read the value from the `WabiSabiConfig` and inject that.